src/ff.rs

```diff
@@ -48,18 +51,32 @@ pub struct FireflyAlgorithm {
   pub config: FireflyAlgorithmCfg,
   pub brightness_function: fn(&Vec<f64>) -> f64,
   pub probe: Box<dyn Probe>,
+  pub distance_function: fn(&Vec<f64>, &[f64]) -> f64,
+}
+
+impl Default for FireflyAlgorithm {
+  fn default() -> Self {
+    FireflyAlgorithm {
+      config: Default::default(),
+      brightness_function: rastrigin,
+      probe: Box::new(StdoutProbe {}),
+      distance_function: cartesian_distance,
+    }
+  }
 }
 
 impl FireflyAlgorithm {
   fn new(
     config: FireflyAlgorithmCfg,
     brightness_function: fn(&Vec<f64>) -> f64,
     probe: Box<dyn Probe>,
+    distance_function: fn(&Vec<f64>, &[f64]) -> f64,
```

Comment:
```suggestion
    distance_function: fn(&[f64], &[f64]) -> f64,
```
It is not necessary, but it its current form it's looking strange to me.